packages/playground/src/components/node_selector/TfSelectionDetails.vue

```diff
@@ -140,6 +145,21 @@ export default {
     const domain = ref<DomainInfo>();
     const domainStatus = ref<ValidatorStatus>();
 
+    const loadedFarms = new Map<number, Farm>();
+    async function loadFarm(farmId: number) {
+      if (loadedFarms.has(farmId)) {
+        return loadedFarms.get(farmId)!;
+      }
+
+      const [farm] = (await gridProxyClient.farms.list({ farmId })).data;
+      loadedFarms.set(farmId, farm);
+      return farm;
```

Comment:
Should return `return loadedFarms.get(farmId);` ?

Reply:
if you mean in getFarm so no we need to keep that sync